app/controllers/katello/api/v2/generic_content_units_controller.rb

```diff
@@ -2,7 +2,8 @@ module Katello
   class Api::V2::GenericContentUnitsController < Api::V2::ApiController
     resource_description do
       name 'Content Units'
-      param :content_type, String, desc: N_("Possible values: #{Katello::RepositoryTypeManager.generic_content_types.join(", ")}"), required: true
+      param :content_type, lambda { |val| Katello::RepositoryTypeManager.generic_content_types.include?(val) },
+            desc: N_("Possible values: %s") % Katello::RepositoryTypeManager.generic_content_types.join(", "), required: true
```

Comment:
While this fixes the translation issue, I don't think this solves @evgeni's issue. You will still call `Katello::RepositoryTypeManager.generic_content_types` on start up. I don't think you can make the description lazy loaded, though it may be possible with a [custom validator](https://github.com/Apipie/apipie-rails?tab=readme-ov-file#adding-custom-validator).

Reply:
This does not fix https://projects.theforeman.org/issues/37977?issue_count=56&issue_position=1&next_issue_id=37976, I just wanted to mention that it was somewhat related.